### DISCUSSION TITLE
Encountering -32602 and -32603 Errors When Fetching Transaction Receipts with Blockchain Indexer

### DISCUSSION THREAD

**Discussion by crypto-le** [blockscout/blockscout#9307] [General]
![119a6ee54fd049e6a6f1dc211c9dc16](https://github.com/blockscout/blockscout/assets/29742985/cf828979-16ac-4f2b-ad61-ec47a08f1334)

I am trying to fetch transaction receipts using a blockchain indexer from an Ethereum node, but I'm encountering some issues. Specifically, when attempting to retrieve the transaction receipts for blocks #2626711 to #2626703, the indexer reports the following errors:

A -32602 error indicating that the requested block hash could not be found on the node, which seems to be a JSON-RPC parameter error.
A -32603 error indicating an internal JSON-RPC error but without providing further context.
Here are the specific error messages I received:
`[error] failed to fetch: [%{code: -32602, data: %{block_hash: "0xda61be527aacfbfd3b3d89bd426737f55a86ef8bc18d94cb6f052166ba4c49be", ...}, message: "Not Found"}]
[error] failed to fetch: [%{code: -32603, data: %{block_hash: "0xda61be527aacfbfd3b3d89bd426737f55a86ef8bc18d94cb6f052166ba4c49be", ...}, message: "Internal JSON-RPC error"}]
`
I notice that the indexer retries and checks at intervals whether it needs to catch up with indexing.

My questions are:

Do these errors suggest that my node is not fully synced, or could it be other configuration or network issues?
What are the common causes for code: -32602, and how should I address this?
What does code: -32603 indicate, and what information do I need to debug this problem more effectively?
Are there recommended steps to diagnose and resolve these types of indexer issues?
Any help or guidance would be greatly appreciated.

**Comment by ekawibisono** [comment]
Hello, can you send detail 
Your RPC Node use : (geth/besu/nethermind/erigon/openethereum)
Blockscout Version : ?

**Reply by crypto-le** [reply]
thank you very much for your prompt and detailed response. I greatly appreciate the time and effort you've taken to address my query regarding the issue I'm experiencing with BlockScout. Based on the guidance you provided, I have conducted further tests and would like to share some additional information for clarity and to seek further assistance.

to recap, my setup was pulled from the master branch around June 2023, and it's currently running on my private blockchain. While RPC tests for blockchain height retrieval are successful, I'm still encountering error messages in the BlockScout logs, which seem inconsistent with the RPC test outcomes.

could you kindly offer further insights or suggest any specific steps I might take to resolve these error messages? Any additional advice or recommendations would be immensely valuable.

thank you once again for your support and patience. I look forward to your expert advice.

**Reply by ekawibisono** [reply]
Could you please provide the details first? What ethereum node are you using? And are you using blockscout version v6.0.0?

**Reply by crypto-le** [reply]
I am currently utilizing an EVMOS node for my project. As you may know, EVMOS is built on Cosmos and supports EVM compatibility. This detail might be crucial as the EVM RPC commands used with EVMOS are identical to those used in Ethereum, which could potentially influence the situation at hand.

Regarding the BlockScout version, there's a possibility that it might be V6.0.0, although I cannot confirm this with absolute certainty. I understand this information might be critical for diagnosing the issue accurately.
